base/abstractarray.jl

```diff
@@ -676,7 +696,7 @@ function hcat{T}(A::AbstractVecOrMat{T}...)
     for j = 1:nargs
         Aj = A[j]
         if size(Aj, 1) != nrows
-            throw(ArgumentError("number of rows must match"))
+            throw(ArgumentError("number of rows of A[$j] ($(size(Aj,1))) must match $nrows"))
```

Comment:
Same here, I think you can say `number of rows of arrays must match (got $(map(x->size(x, 1), A)))`.